Frends.PowerShell.RunCommand/Frends.PowerShell.RunCommand/PowerShell.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.ComponentModel;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Runtime.CompilerServices;
+
+#pragma warning disable 1591
+
+[assembly: InternalsVisibleTo("Frends.PowerShell.RunCommand.Tests")]
+namespace Frends.PowerShell.RunCommand
+{
+    public static class PowerShell
+    {
+        /// <summary>
+        /// Creates a PowerShell session which can be shared between scripts and commands.
+        /// </summary>
+        /// <returns>Object</returns>
+        internal static SessionWrapper CreateSession()
+        {
+            // Leave this internal for now, turn public to support shared sessions between task executions
+            return new SessionWrapper();
+        }
+
+        private static PowerShellResult DoAndHandleSession(SessionWrapper sessionFromOutside, Func<SessionWrapper, PowerShellResult> action)
+        {
+            SessionWrapper internalSession = null;
+            // use the external session if provided or create and dispose an internal session
+            var session = sessionFromOutside ?? (internalSession = CreateSession());
+
+            try
+            {
+                return action.Invoke(session);
+            }
+            finally
+            {
+                internalSession?.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Executes a PowerShell command with parameters, leave parameter value empty for a switch
+        /// </summary>
```

Comment:
Also add `<param name="input">documentation</param>` to document input and options parameters.

Reply:
Fixed in both RunCommand and RunScript tasks.